### PR TITLE
Performance improvements from Forest Fang, by not doing a full transf…

### PR DESCRIPTION
**Pull request by jq**
…orm on all samples

@saurfang @deerzq

### CODE REVIEW

File: training/src/main/scala/com/airbnb/aerosolve/training/AdditiveModelTrainer.scala [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package com.airbnb.aerosolve.training
+package com.airbnb.marketplace.pricing
```
Comment (deerzq) [inline training/src/main/scala/com/airbnb/aerosolve/training/AdditiveModelTrainer.scala]: pricing?

Comment (jq) [inline training/src/main/scala/com/airbnb/aerosolve/training/AdditiveModelTrainer.scala]: nice catch, updated!

**Comment by saurfang** [comment]
Added more documentation around the motivation and caveat of early sampling. Added wrapper functions for backwards compatibility. Removed duplicated sampling in model initialization. 
@jq Did you mean to sample *with* replacement in NDTree or was it an oversight? (This changes it to sample without replacement)

**Comment by jq** [comment]
@saurfang I pass in true since it doesn't need to be remember sampled for NDTree, a slightly better performance?

**Comment by jq** [comment]
👍  let's get it in and try

**Comment by deerzq** [comment]
LGTM

File: training/src/main/scala/com/airbnb/aerosolve/training/AdditiveModelTrainer.scala [hunk 2]
```diff
@@ -53,15 +53,19 @@ object AdditiveModelTrainer {
                                    linearFeatureFamilies: java.util.List[String],
                                    priors: Array[String],
                                    nDTreeBuildOptions: NDTreeBuildOptions)
+  def train(sc: SparkContext, input: RDD[Example], config: Config, key: String): AdditiveModel =
+    train(sc, (frac: Double) => input.sample(false, frac), config, key)
```
Comment (yinlou) [inline training/src/main/scala/com/airbnb/aerosolve/training/AdditiveModelTrainer.scala]: Why do we need this? I thought we subsampled before calling this train?

Comment (saurfang) [inline training/src/main/scala/com/airbnb/aerosolve/training/AdditiveModelTrainer.scala]: For future reference, we want to sample during each iteration and initialization. We prefer not to double sample during model training.

**Pull request merged**